### PR TITLE
Paged iterable support paged flux base

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: sdk/keyvault/azure-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ protected void beforeTest() {
                 .pipeline(pipeline)
                 .endpoint(getEndpoint())
                 .buildAsyncClient());
-        }
```
Comment (JonathanGiles) [inline sdk/keyvault/azure-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java]: This change can be reverted :-)

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java [hunk 2]
```diff
@@ -56,17 +57,19 @@ public PagedIterable(PagedFlux<T> pagedFlux) {
      *
      * @return {@link Stream} of {@link PagedResponse}, starting from the page associated with the continuation token
      */
-    public Stream<PagedResponse<T>> streamByPage(String continuationToken) {
-        return pagedFlux.byPage(continuationToken).toStream();
+    @SuppressWarnings("unchecked")
+    public <P extends  PagedResponse<T>> Stream<P> streamByPage(String continuationToken) {
```
Comment (JonathanGiles) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]:
```suggestion
    public <P extends PagedResponse<T>> Stream<P> streamByPage(String continuationToken) {
```

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java [hunk 3]
```diff
@@ -44,8 +44,9 @@ public PagedIterable(PagedFlux<T> pagedFlux) {
      * It will provide same {@link Stream} of T values from starting if called multiple times.
      * @return {@link Stream} of {@link PagedResponse}
      */
-    public Stream<PagedResponse<T>> streamByPage() {
-        return pagedFlux.byPage().toStream();
+    @SuppressWarnings("unchecked")
+    public <P extends PagedResponse<T>> Stream<P> streamByPage() {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: Update the javadocs also to reflect the changes in return types.

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java [hunk 4]
```diff
@@ -23,18 +23,18 @@
...
 public class PagedIterable<T> extends IterableStream<T> {
-    private final PagedFlux<T> pagedFlux;
+    private final PagedFluxBase<T, ? extends PagedResponse<T>> pagedFlux;
 
     /**
      * Creates instance given {@link PagedFlux}.
      * @param pagedFlux to use as iterable
      */
-    public PagedIterable(PagedFlux<T> pagedFlux) {
+    public <P extends PagedResponse<T>> PagedIterable(PagedFluxBase<T, P> pagedFlux) {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: Update javadocs to reflect the new input and output types.

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java [hunk 5]
```diff
@@ -23,18 +23,18 @@
  *
  * {@codesnippet com.azure.core.http.rest.pagedIterable.iterableByPage.while}
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: Code snippets should include examples for `PagedFlux` and `CustomPagedFlux` extending from `PagedFluxBase`.

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java [hunk 6]
```diff
@@ -23,18 +23,18 @@
...
  *
- * @param  <T> The type of value contained in this {@link IterableStream}.
+ * @param <T> The type of value contained in this {@link IterableStream}.
  * @see PagedResponse
  * @see IterableStream
  */
 public class PagedIterable<T> extends IterableStream<T> {
-    private final PagedFlux<T> pagedFlux;
+    private final PagedFluxBase<T, ? extends PagedResponse<T>> pagedFlux;
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: Update name to `pagedFluxBase`

File: sdk/keyvault/azure-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java (hunk 1 above)
Comment (srnagar) [reply sdk/keyvault/azure-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java]: 👍

Comment (g2vinay) [reply sdk/keyvault/azure-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java]: It was added to test SDK with PagedIterable change

Comment (g2vinay) [reply sdk/keyvault/azure-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java]: removed now.

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java (hunk 2 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: done

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java (hunk 3 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: updated

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java (hunk 4 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: updated

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java (hunk 5 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: Added

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java (hunk 6 above)
Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java]: updated

**Pull request merged**